### ISSUE TITLE
Bug: blank screen on database connect after OS update

### ISSUE THREAD

**Issue by simeonnv** [surrealdb/surrealist#424]
### Describe the bug

surrealist runs normally till i try to connect to my server.
the screen goes blank and the console stops outputing

### Steps to reproduce

im on linux mint fresh out of the box
im using `WEBKIT_DISABLE_DMABUF_RENDERER=1 surrealist` to run it

### Expected behaviour

i connect to my database and surrealist stops responding to my mouse and clicks
and after some time it goes blank

the console output is
https://pastebin.com/BUSEv10T

### Surrealist Environment

surrealist v3.0.0-beta.2 or v2.1.6 (these are the ones i tried)

### Additional info
this bug started to occur after a linux mint update

**Comment by shadowInvert** [comment]
Sorry, my English isn’t very good.

### Describe the bug
surrealist runs normally till i try to connect to my server.
the screen goes blank and the console stops outputing

### Steps to reproduce
I’m ubuntu on wsl2
im using WEBKIT_DISABLE_DMABUF_RENDERER=1 


### Surrealist Environment
surrealist v2.1.5

### Additional info
this bug started to occur after
In the query input field of Surrealist, I entered the following query:
update register:xxxxxxxxx set attention += 'new-email@example.com';

### What I tried to do to solve it
Updated from ver2.1.5 to ver2.1.6
Delete Surrealist ver2.1.6 and reinstall it.

**Comment by macjuul** [comment]
@simeonnv @shadowInvert Could you check if the [developer console](https://surrealdb.com/docs/surrealist/troubleshooting#opening-the-developer-console) contains any errors or other hints as to what could be causing this?

**Comment by macjuul** [comment]
Just for confirmation here, the latest stable release (2.1.6) was working correctly, however it suddenly stopped working after an operating system update?

**Comment by Mauge9638** [comment]
> Just for confirmation here, the latest stable release (2.1.6) was working correctly, however it suddenly stopped working after an operating system update?

I don't know about the others, but yes that's what happened to me, had an update on Ubuntu 24.04 yesterday, and now it doesn't work. (Same issues as stated by the others)

**Comment by simeonnv** [comment]
> Just for confirmation here, the latest stable release (2.1.6) was working correctly, however it suddenly stopped working after an operating system update?

no it doesn't. I tried it

**Comment by macjuul** [comment]
Is this happening with the `.deb`, `.rpm`, or `.AppImage` installers?

If deb or rpm was used, would it be possible to try the AppImage instead?

**Comment by anmwinter** [comment]
For me it's with the .deb files. I'll try the AppImage.

**Comment by anmwinter** [comment]
For 2.16 the AppImage runs fine on Pop_OS/Ubuntu but not the .deb.
3.0 beta AppImage loads but when you select Connect it hangs and then you get the white screen of doom.

**Comment by simeonnv** [comment]
appimage 2.1.6 works fine on mint

**Comment by shadowInvert** [comment]
I think I was using version 2.1.5.But it may have been downloaded automatically
The situation where this happened to me was when I tried adding an array to a field in the Surrealist command line, checked it, and then launched Surrealist again about 10 minutes later.

For me it's with the .deb files.



This is the state of the system when Surrealist was launched.
<details>
<summary>Expand</summary>
```
journalctl -xe

9月 11 19:24:08 SHADOW kernel: Code: 41 89 c5 41 f7 dd eb 80 66 0f 1f 44 00 00 b8 ba 00 00 00 0f 0>
 9月 11 19:24:08 SHADOW kernel: RSP: 002b:00007ee79c7f3100 EFLAGS: 00000246 ORIG_RAX: 0000000000000>
 9月 11 19:24:08 SHADOW kernel: RAX: 0000000000000000 RBX: 00007ee79c7fc640 RCX: 00007f002980d9fc
 9月 11 19:24:08 SHADOW kernel: RDX: 0000000000000006 RSI: 0000000000003fc0 RDI: 0000000000003f9d
 9月 11 19:24:08 SHADOW kernel: RBP: 0000000000003fc0 R08: 00007ee79c7f31d0 R09: 00007ee79c7f3300
 9月 11 19:24:08 SHADOW kernel: R10: 0000000000000008 R11: 0000000000000246 R12: 0000000000000006
 9月 11 19:24:08 SHADOW kernel: R13: 0000000000000016 R14: 00000000000000ff R15: 00007ee79c7f4770
 9月 11 19:24:08 SHADOW kernel: FS:  00007ee79c7fc640 GS:  0000000000000000
 9月 11 19:25:54 SHADOW kernel: misc dxg: dxgk: dxgkio_is_feature_enabled: Ioctl failed: -22
 9月 11 19:25:54 SHADOW kernel: misc dxg: dxgk: dxgkio_query_adapter_info: Ioctl failed: -22
 9月 11 19:25:54 SHADOW kernel: misc dxg: dxgk: dxgkio_query_adapter_info: Ioctl failed: -22
 9月 11 19:25:54 SHADOW kernel: misc dxg: dxgk: dxgkio_query_adapter_info: Ioctl failed: -22
 9月 11 19:25:54 SHADOW kernel: misc dxg: dxgk: dxgkio_query_adapter_info: Ioctl failed: -2
 9月 11 19:25:54 SHADOW kernel: misc dxg: dxgk: dxgkio_query_adapter_info: Ioctl failed: -22
 9月 11 19:25:54 SHADOW kernel: misc dxg: dxgk: dxgkio_query_adapter_info: Ioctl failed: -22
 9月 11 19:25:54 SHADOW kernel: misc dxg: dxgk: dxgkio_query_adapter_info: Ioctl failed: -22
 9月 11 19:25:54 SHADOW kernel: misc dxg: dxgk: dxgkio_query_adapter_info: Ioctl failed: -2
 9月 11 19:25:54 SHADOW rtkit-daemon[373]: Supervising 0 threads of 0 processes of 0 users.
 9月 11 19:25:54 SHADOW kernel: potentially unexpected fatal signal 6.
 9月 11 19:25:54 SHADOW kernel: CPU: 8 PID: 20101 Comm: t Helper Thread Not tainted 5.15.153.1-micr>
 9月 11 19:25:54 SHADOW kernel: RIP: 0033:0x7f167e4a29fc
 9月 11 19:25:54 SHADOW kernel: Code: 41 89 c5 41 f7 dd eb 80 66 0f 1f 44 00 00 b8 ba 00 00 00 0f 0>
 9月 11 19:25:54 SHADOW kernel: RSP: 002b:00007f15eeff0100 EFLAGS: 00000246 ORIG_RAX: 0000000000000>
 9月 11 19:25:54 SHADOW kernel: RAX: 0000000000000000 RBX: 00007f15eeff9640 RCX: 00007f167e4a29fc
 9月 11 19:25:54 SHADOW kernel: RDX: 0000000000000006 RSI: 0000000000004210 RDI: 00000000000041e9
 9月 11 19:25:54 SHADOW kernel: RBP: 0000000000004210 R08: 00007f15eeff01d0 R09: 00007f15eeff0300
 9月 11 19:25:54 SHADOW kernel: R10: 0000000000000008 R11: 0000000000000246 R12: 0000000000000006
 9月 11 19:25:54 SHADOW kernel: R13: 0000000000000016 R14: 00000000000000ff R15: 00007f15eeff1770
 9月 11 19:25:54 SHADOW kernel: FS:  00007f15eeff9640 GS:  0000000000000000
```
</details>

**Comment by Mauge9638** [comment]
Appimage 3.0 beta 1 works fine on ubuntu 24.04 (haven't tried others)

**Comment by ansarizafar** [comment]
Appimage 3.0 beta 1 works fine on ubuntu 22.04 with SurrealDB 2.0 but appimage 3.0 beta 2 shows blank screen when I try to connect to the database.

**Comment by dayo777** [comment]
I am on Ubuntu 24.04.1, and can confirm this same issue. After updating from 24.04, surrealist stopped working. 

The App opens but just goes blank. I uninstalled Surrealist & reinstalled but still experiencing the issue.

**Comment by simeonnv** [comment]
all versions of surrealist after 3.0.0 beta 2 has this issue with the appimages.

**Comment by macjuul** [comment]
Hey everyone, we certainly noticed the significance of this issue and are continuing to look for a solution. We cannot reproduce this issue internally with the devices to our disposal, which has not made it trivial to debug. Additionally the circumstances causing this issue appear to vary greatly, making it additionally difficult to pin down. What we do know is that it only affects Linux devices.

To move things forward I have some additional questions I would like to gather feedback on.

- Please try opening the developer tools panel, for which an explanation can be found [here](https://surrealdb.com/docs/surrealist/troubleshooting#opening-the-developer-console). This may lead to further details as to what causes this issue, especially within the Console tab.
- Does this only affects the desktop app or also the web app? Is this consistent across different browsers?
- Does backing up and deleting the config file (`$HOME/.config/SurrealDB/Surrealist/config.json`) solve the issue?
- Is there a reproducible way to trigger the blank screen?

Hopefully additional insight might lead us closer to the root cause so we can fix this. Thank you all for your help and patience!

**Comment by dayo777** [comment]
@macjuul 
OS => Ubuntu 24.04.1 LTS  (gnome 46)
Kernel version => Linux 6.8.0-45-generic

1. I tried opening the developer panel on Linux using `CTRL + shift + i` but nothing happens. Tried it multiple times, nothing happens still. 
2. This only affects the Desktop app, web app works fine. 
3. I deleted the config file, but issue still persists.
4. Not sure about this. I have included my OS & kernel version, just in-case. Only thing I can think of is trying this out in a VirtualBox.

**Comment by hollygrimm** [comment]
I'm also getting a blank screen on my Linux desktop app after the upgrade from 2.1.5 to 3.0.2

I deleted the `.config/SurrealDB/Surrealist` folder and reinstalled the .deb so I could at least see the startup screen. I'm able to bring up `Developer tools` here. No errors at this point.

`Create Connection` As soon as I try to create a ws or http connection to my localhost, the screen goes blank and the Developer tools no longer display so I can't see any errors. I'm using the same connection on my command line and that works fine.

`Open the Sandbox` is also blank screen.

No errors in `surrealist.log`.

**Comment by shadowInvert** [comment]
I use WSL2

cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.5 LTS"

 ## What I did before it happened
I ran the following query in Surrealist:
update register:xxxxxxxxx set attention += '[new-email@example.com]';

## What I did after it happened
1. sudo dpkg -r surrealist  -> I downloaded ver2.1.6  -> blank screen
2. sudo dpkg -r surrealist  ->  I deleted the .config/SurrealDB/Surrealist  ->  I downloaded ver2.1.6  -> blank screen
3. sudo dpkg --purge surrealist -> sudo apt-get autoremove  ->
     I deleted the .config/SurrealDB/Surrealist  and  .cache/SurrealDB/Surrealist.  -> I downloaded ver2.1.6  -> blank screen

## What  I noticed
Shortcuts not working properly after re-downloading the Surrealist_2.1.6_amd64.deb

## now
 appimage 2.1.6 works fine

**Comment by gander** [comment]
@macjuul 

| what       | ver              |
|------------|------------------|
| Linux Mint | 22               |
| Cinnamon   | 6.2.9            |
| Ubuntu     | 23.04            |
| Kernel     | 6.8.0-45-generic |

> Please try opening the developer tools panel, for which an explanation can be found [here](https://surrealdb.com/docs/surrealist/troubleshooting#opening-the-developer-console). This may lead to further details as to what causes this issue, especially within the Console tab.

nothing is happening

> Does this only affects the desktop app or also the web app? Is this consistent across different browsers?

only desktop versions. all PWA and Web solutions work fine.

> Does backing up and deleting the config file ($HOME/.config/SurrealDB/Surrealist/config.json) solve the issue?

nope

> Is there a reproducible way to trigger the blank screen?

Try fresh install on Linux Mint 22 Cinnamon

**Comment by dayo777** [comment]
On Surrealist 3.0.3, the most recent version, this issue still exists. I have tried both the *.deb & AppImage, same!

I'm running GNOME 46 on Ubuntu 24.04.1 LTS.

Issue:

Upon opening the app, the screen becomes completely blank or stops working.

**Comment by ovniroto** [comment]
I have the same problem on Fedora 40 with GNOME 46. With both AppImage and rpm, the whole Surrealist interface gets very slow, or shows a blank screen.

**Comment by IDEDARY** [comment]
Same issue on Linux mint with any version past 3.0.0. To put salt in the wound, rolling back to older releases stopped working after I opened the new version once. The versions that worked before now do not work even after clearing cache and reinstall (I did not update the OS in the meantime).

Related: https://github.com/surrealdb/surrealist/issues/492, but then I got stuck here too.

**Comment by Mauge9638** [comment]
If some people here can't get it to work with different versions try AppImage "Surrealist 3.0.0", it works for me right now as a workaround. I'm on Ubuntu 24.04.1 LTS , Gnome 46. It works both on X11 and Wayland.

**Comment by gander** [comment]
I can confirm that the only `3.x` version that works for me is `beta1`. Whatever broke happened between `beta1` and `beta2`.

Linux Mint 22, RTX on board (I saw somewhere that this might be a factor)

**Comment by Mauge9638** [comment]
> I can confirm that the only `3.x` version that works for me is `beta1`. Whatever broke happened between `beta1` and `beta2`.
> 
> Linux Mint 22, RTX on board (I saw somewhere that this might be a factor)

Might be worth noting i'm on RTX 2080 aswell, and if relevant AMD Ryzen 7 7700X

**Comment by ovniroto** [comment]
I confirm that beta1 also works for me (AppImage on Fedora 40).

**Comment by gander** [comment]
https://github.com/surrealdb/surrealist/issues/276

**Comment by agarfer1** [comment]
I have been able to reproduce this error down to version 2.1.0 (I guess that's the first version that updated to webkit2gtk-4.1?)

1. The developer tools go blank as well.
2. Only tried the desktop app.
3. Deleting the config.json information on queries and queryHistory lets me go back to using the app.
4. Don't know how to reproduce it, but this happened from scratch on an Ubuntu 24.04 fresh install that "inherited" a working config.json from Surrealist 2.0.6

Platform: Desktop
OS: linux
Architecture: x86_64
WebView: Mozilla/5.0 (X11; Ubuntu; Linux x86_64) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.0 Safari/605.1.15
Version: 2.1.0
Flags: featureFlags: false, models_view: true, apidocs_view: true, themes: false, newsfeed: true, database_version_check: true

**Comment by macjuul** [comment]
Hello everyone! I know this has been a frustrating problem for all of you, and while I do not bring a solution yet, I do have some findings to report.

After extensive testing on Windows through WSL 2 Running Ubuntu 22 I have been able to successfully reproduce the crash. I was able to narrow down the cause of this issue to the WebAssembly used by Surrealist, which includes the surrealdb runtime (@surrealdb/wasm) and the SurrealQL support package (@surrealdb/ql-wasm).

Executing this WebAssembly code causes the browser process used by Surrealist (webkit2gtk) to freeze and crash entirely. This is caused by [a recent issue in webkit2gtk-4.1](https://bugs.launchpad.net/ubuntu/+source/webkit2gtk/+bug/2081166) which made it to the live package. Unfortunately at this time it does not appear a solution is available yet.

I hope this gives some context and perspective on the issue at hand, and while we cannot solve it at this time as it is technically caused by an issue external to Surrealist, I will continue to closely monitor and report back on any updates.

Thank you all for your patience!

**Comment by liusong1111** [comment]
I found it always visit `https://api.cloud.surrealdb.com`, and I don't know whether it helps.


1. "New Connection", fill "localhost:8000"
2. After Click "Create", it's blocked and soon become blank white.

tail `$HOME/.config/SurrealDB/Surrealist/logs/surrealist.log`:

```
2024-10-14 03:45:52 [INFO] Fetching requested resources
2024-10-14 03:45:52 [INFO] Updater: Checking for updates
2024-10-14 03:45:52 [TRACE] checkout waiting for idle connection: ("https", surrealist.app)
2024-10-14 03:45:52 [DEBUG] starting new connection: https://surrealist.app/
2024-10-14 03:45:52 [TRACE] Http::connect; scheme=Some("https"), host=Some("surrealist.app"), port=None
2024-10-14 03:45:52 [DEBUG] resolving host="surrealist.app"
2024-10-14 03:45:52 [DEBUG] connecting to 13.35.210.68:443
2024-10-14 03:45:52 [DEBUG] connected to 13.35.210.68:443
2024-10-14 03:45:52 [TRACE] checkout waiting for idle connection: ("https", api.cloud.surrealdb.com)
2024-10-14 03:45:52 [DEBUG] starting new connection: https://api.cloud.surrealdb.com/
2024-10-14 03:45:52 [TRACE] Http::connect; scheme=Some("https"), host=Some("api.cloud.surrealdb.com"), port=None
2024-10-14 03:45:52 [DEBUG] resolving host="api.cloud.surrealdb.com"
2024-10-14 03:45:52 [DEBUG] connecting to 107.23.205.14:443
2024-10-14 03:45:52 [TRACE] http1 handshake complete, spawning background dispatcher task
2024-10-14 03:45:52 [TRACE] checkout dropped for ("https", surrealist.app)
2024-10-14 03:45:52 [DEBUG] connected to 107.23.205.14:443
2024-10-14 03:45:53 [TRACE] http1 handshake complete, spawning background dispatcher task
2024-10-14 03:45:53 [TRACE] checkout dropped for ("https", api.cloud.surrealdb.com)
2024-10-14 03:45:53 [TRACE] put; add idle connection for ("https", surrealist.app)
2024-10-14 03:45:53 [DEBUG] pooling idle connection for ("https", surrealist.app)
2024-10-14 03:45:53 [INFO] Updater: No updates available
2024-10-14 03:45:53 [TRACE] put; add idle connection for ("https", api.cloud.surrealdb.com)
2024-10-14 03:45:53 [DEBUG] pooling idle connection for ("https", api.cloud.surrealdb.com)
2024-10-14 03:45:53 [INFO] Cloud: Refreshing cloud access token
2024-10-14 03:45:53 [INFO] Cloud: Invalidating active session     <<<--- The Last Line
```

**Comment by shadowInvert** [comment]
use ubuntu22.04 on WSL2
use Surrealist_2.1.6_amd64.AppImage

There was a notification in the bottom left of the Surrrealist app notifying me of an update to 3.0.8version, so I cautiously ran it and it became balnk.
Now I can't use the AppImage or deb files.


Can someone tell me how to delete all related files and start over from scratch after the app crashes?
I delete Surrealist/config.json and run it again, this just opens.↓
![ 2024-10-16](https://github.com/user-attachments/assets/5581ead6-f091-4006-9b67-73bd8feac90c)

**Comment by koakh** [comment]
I confirm that this problem don't exists in version Surrealist_3.0.0-beta.1_amd64.appimage on NixOS
all versions above this will freeze on startup
something happens from Surrealist_3.0.0-beta.1_amd64.appimage to Surrealist_3.0.0-beta.2_amd64.appimage

**Comment by Mauge9638** [comment]
> use ubuntu22.04 on WSL2 use Surrealist_2.1.6_amd64.AppImage
> 
> There was a notification in the bottom left of the Surrrealist app notifying me of an update to 3.0.8version, so I cautiously ran it and it became balnk. Now I can't use the AppImage or deb files.
> 
> Can someone tell me how to delete all related files and start over from scratch after the app crashes? I delete Surrealist/config.json and run it again, this just opens.↓ ![ 2024-10-16](https://private-user-images.githubusercontent.com/126877329/376927257-5581ead6-f091-4006-9b67-73bd8feac90c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkwNzgzODksIm5iZiI6MTcyOTA3ODA4OSwicGF0aCI6Ii8xMjY4NzczMjkvMzc2OTI3MjU3LTU1ODFlYWQ2LWYwOTEtNDAwNi05YjY3LTczYmQ4ZmVhYzkwYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxNlQxMTI4MDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZTgwMzJlMWNhMDNiYWY3YmZhNjg3NDQwZDUyOGM2NzdjYTg0ODgyYzVlYjQyNmRlYTNlYWE5NGJmNTRkZTE4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.j2-cpGbZAj98hBn-Tj4aK-drsPH1cen4Cz_ZWLmj5FI)

Have you tried to download the working AppImage again? And then use the newly downloaded AppImage onward (or until Surrealist implements a fix)

**Comment by gander** [comment]
This is the same problem over and over again, and it is unlikely that anyone will be able to solve it because: https://github.com/surrealdb/surrealist/issues/424#issuecomment-2407484597

**Comment by agarfer1** [comment]
Surrealist seems to be working as expected now, after I installed an update in the libwebkit2gtk-4.1-0 library for Ubuntu 24.04.

Can somebody else confirm if `sudo apt update && sudo apt upgrade` fixed this error for them?

**Comment by Mauge9638** [comment]
> Surrealist seems to be working as expected now, after I installed an update in the libwebkit2gtk-4.1-0 library for Ubuntu 24.04.
> 
> Can somebody else confirm if `sudo apt update && sudo apt upgrade` fixed this error for them?

Seems like it works for me aswell (after i update and upgrade), will do more testing of it tomorrow, but if i don't edit this comment by tomorrow it works again

**Comment by gander** [comment]
I confirm that updating the `libwebkit2gtk-4.1-0` package solved my problems on `Linux Mint 22` (`6.8.0`). I installed the latest version of the application from the file `Surrealist_3.0.8_amd64.deb` and it works without problems.

🥳

**Comment by macjuul** [comment]
After testing myself I can indeed confirm the latest version of `libwebkit2gtk` is working and Surrealist is once again compatible. 🎉

For those looking to resolve this issue, simply upgrade your packages and the current installed version of Surrealist should work again.
```
sudo apt update && sudo apt upgrade
```
